cmd/query/app/server.go

```diff
@@ -346,6 +356,16 @@ func (s *Server) Start() error {
 
 // Close stops http, GRPC servers and closes the port listener.
 func (s *Server) Close() error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if !s.started {
+		return nil
+	}
+	s.started = false
+
+	s.logger.Info("Stopping server")
+	close(s.stopChan)
```

Comment:
Nobody reads this channel

Reply:
I appreciate the feedback. Perhaps logging directly after the close operation would be more effective.